public/apps/configuration/utils/request-utils.ts

```diff
@@ -13,29 +13,85 @@
  *   permissions and limitations under the License.
  */
 
-import { HttpStart, HttpHandler } from 'opensearch-dashboards/public';
+import { HttpStart, HttpHandler, HttpFetchQuery } from 'opensearch-dashboards/public';
 
-export async function request<T>(requestFunc: HttpHandler, url: string, body?: object): Promise<T> {
-  if (body) {
-    return (await requestFunc(url, { body: JSON.stringify(body) })) as T;
-  }
-  return (await requestFunc(url)) as T;
+interface RequestType {
+  http: HttpStart;
+  url: string;
+  body?: object;
+}
+
+interface RequestParams {
+  requestFunc: HttpHandler;
+  url: string;
+  body?: object;
+  query: HttpFetchQuery;
 }
 
-export async function httpGet<T>(http: HttpStart, url: string): Promise<T> {
-  return await request<T>(http.get, url);
+export function createRequestContextWithDataSourceId(dataSourceId: string) {
+  if (dataSourceId === undefined) {
+    throw new Error('dataSourceId is not present');
+  }
+  return new RequestContext(dataSourceId);
 }
 
-export async function httpPost<T>(http: HttpStart, url: string, body?: object): Promise<T> {
-  return await request<T>(http.post, url, body);
+export class RequestContext {
+  query: HttpFetchQuery;
+  constructor(private readonly dataSourceId: string) {
+    this.query = {
+      dataSourceId: this.dataSourceId,
+    };
+  }
+
+  public async httpGet<T>(params: RequestType): Promise<T> {
+    const { http, url, body } = params;
+    return await request<T>({ requestFunc: http.get, url, body, query: this.query });
```

Comment:
It actually can't - I tried. Since query is a member of the class, we need to either extract it out the line before via `const query = this.query` if we want to do it that way, but I think this works and is pretty clear.